devel/namespaces/index.md

```diff
@@ -0,0 +1,169 @@
+---
+layout:   default
+class:    markdown
+---
+
+Namespaces
+==========
+
+TaskCluster has a number of namespaces defined to allow multiple users to get along without interfering with one another.
+The platform itself is agnostic to the structure of these namespaces, but infrastructure that interacts with the platform is dependent on the namespaces for security and correctness.
+
+This document necessarily contains information that is specific to users of the TaskCluster platform.
+As such, it is open to contributions from all users who wish to carve out a section of a namespace -- just [submit a pull request](https://github.com/taskcluster/taskcluster-docs).
+
+## Projects
+
+Most work at Mozilla falls into "projects", and these provide a nice organizational boundary for controlling access.
+We use a consistent name for each project in the various namespaces below -- something simple and without punctuation.
+The known projects are:
+
+ * `taskcluster` -- The taskcluster platform itself
+ * `releng` -- Mozilla release engineering (build, test, and release processes)
+ * `gaia` -- B2G/Firefox OS build system
+ * `bmo` -- bugzilla.mozilla.org development
+
+Please file a pull request to add your project!
+
+## Scopes
+
+Many scopes reflect the namespaces given elsewhere in this document, as described in the API documentation for the component.
+
+* `<component>:<action>:<resource>`,
+* `<component>:<action>:<qualifier>`,
+* `<component>:<action>:<specifier>` -
+   Scopes for most API actions follow this pattern.
+   For example, the `queue.createTask` call is governed by a scope beginning with `queue:create-task:<provisionerId>/<workerType>`.
```

Comment:
`queue.defineTask` and by `queue:define-task:...` is a better more long term example.